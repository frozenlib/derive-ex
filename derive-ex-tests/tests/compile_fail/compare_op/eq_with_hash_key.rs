use derive_ex::derive_ex;

#[derive_ex(Hash, Eq, PartialEq)]
struct X(#[hash(key = $.len())] String);

fn main() {}
