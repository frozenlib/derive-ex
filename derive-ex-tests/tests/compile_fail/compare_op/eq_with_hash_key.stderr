error: Since `#[hash(key = ...)]` was specified, the default implementation of `Eq` cannot be used.
       One of the following attributes is required.

       #[eq(key = ...)]
       #[eq(by = ...)]
       #[ord(key = ...)]
       #[ord(by = ...)]
 --> tests/compile_fail/compare_op/eq_with_hash_key.rs:4:23
  |
4 | struct X(#[hash(key = $.len())] String);
  |                       ^

error: Since `#[hash(key = ...)]` was specified, the default implementation of `PartialEq` cannot be used.
       One of the following attributes is required.

       #[partial_eq(key = ...)]
       #[partial_eq(by = ...)]
       #[eq(key = ...)]
       #[eq(by = ...)]
       #[partial_ord(key = ...)]
       #[partial_ord(by = ...)]
       #[ord(key = ...)]
       #[ord(by = ...)]
 --> tests/compile_fail/compare_op/eq_with_hash_key.rs:4:23
  |
4 | struct X(#[hash(key = $.len())] String);
  |                       ^
